### DISCUSSION TITLE
Creating a Stable Buildroot Linux Image for v86

### DISCUSSION THREAD

**Discussion by cemalgnlts** [copy/v86#872] [Q&A]
Hi,

I'm trying to build a small linux image with Buildroot, some of the code comes from [humphd/browser-vm](https://github.com/humphd/browser-vm) and some from the [SkiffOS](https://github.com/skiffos/SkiffOS/tree/master/configs/browser/v86) project.

In addition I am adding the pre-built nodejs binary. fs9p is mounted in the `/root` folder.

I run the following code 10 times (That way it probably runs faster each time with the cache and I can minimize it):
```bash
node /opt/now/hello.js > /dev/null
```

On call 7th I get the following error:
```diff
[7/10] Node initializing...node:internal/fs/utils:347
    throw err;
    ^

- Error: ENOENT: no such file or directory, lstat '5opt'
    at Object.realpathSync (node:fs:2533:7)
    at toRealPath (node:internal/modules/cjs/loader:466:13)
    at Function.Module._findPath (node:internal/modules/cjs/loader:622:22)
    at resolveMainPath (node:internal/modules/run_main:19:25)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:75:24)
    at node:internal/main/run_main_module:22:47 {
  errno: -2,
  syscall: 'lstat',
  code: 'ENOENT',
-  path: '5opt'
}
```

The problem here is that it converts the `/opt` folder to `5opt`.  If I work in the `/root` folder it now gives this error:
```
Error: ENOENT: no such file or directory, lstat 'uroot'
path: 'uroot'
```

Now it also recognizes the `/root` folder as `uroot`. This problem is only on the nodejs side, nothing has changed on the linux command line. I don't know why it detects path names differently. And this only happens after the 7th run, first time no problem.

The config I used for Buildroot: [cemalgnlts/now-linux-build](https://github.com/cemalgnlts/now-linux-build)

**Comment by copy** [comment]
Could you upload the built image file for me to reproduce (including the contents of the hello.js script)?

**Reply by cemalgnlts** [reply]
Of course I build the iso file using CircleCI (similar to GitHub actions).

~ISO file: [linux.iso](https://app.circleci.com/pipelines/github/cemalgnlts/now-linux-build/80/workflows/4f3e63e4-d876-4567-a0ee-f0ea0ec9d706/jobs/87/artifacts).~
State file: [v86state.bin](https://gofile.io/d/2AjdG9).

This problem does not occur every time. This problem occurred again on my last attempt, so I saved it by saying **Save State**. On the [copy.sh/v86](https://copy.sh/v86/) page, go to **Setup** and initialize **512MB** of space, press **Start Emulation** then install state.

When the boot process is finished, you can run the following command:
```sh
. /opt/now/boot-node.sh
```

You can run the node like this to test it:
```sh
node /opt/now/hello.js
```

**Reply by cemalgnlts** [reply]
Now that I have figured out how to see the error, after saying **Load State**, call the following commands in order:

```sh
npm --version
```

You will see that it takes the path `/usr` as `ousr`.

```sh
node --version
node /opt/now/hello.js
```

And it will recognize `/opt` as `5opt`.
